File: python/word-count/word_count.py
from collections import defaultdict
import re


def word_count(sentence):
    """
    Given a phrase, count the occurrences of each word in that phrase.
    
    :param sentence: string sentence to count
    :return: dictionary of words and counts
    """
    word_counts = defaultdict(int)
    words = re.split(r'[\s\W_]', sentence.lower())
    for word in words:
        if word:
            word_counts[word] += 1
    return word_counts

File: typescript/word-count/word-count.ts
export default class Words {
    count(phrase: string): Object {
        const results: { [s: string]: number; } = {}
        phrase.trim().toLowerCase().split(/[\s]/).filter((s) => s !== '').map((word: string) => {
            results[word] = (results.hasOwnProperty(word)) ? results[word] + 1 : 1
        })
        return new Map(Object.entries(results))
    }
}